File: core/extract/facades/security_facade.py
from core.security.remote_target import is_safe_remote_target
from core.security.resource_budget import enforce_resource_budget
from core.security.hardening import (
    decompression_guard,
    malformed_payload_guard,
    memory_guard,
    recursion_guard,
    redirect_guard,
    sandbox_text,
    ssrf_guard,
    timeout_guard,
)
from core.performance import budgeted_chunks, incremental_parse, lazy_extract, memory_budget, parser_pool, stream_parse

File: src/extract/facades/securityFacade.ts
/**
 * Converted from Python: core/extract/facades/security_facade.py
 * @generated — WebWeaveX python→javascript library port
 */

import { isSafeRemoteTarget } from "../../security/remoteTarget.js";
import { enforceResourceBudget } from "../../security/resourceBudget.js";
import { decompressionGuard, malformedPayloadGuard, memoryGuard, recursionGuard, redirectGuard, sandboxText, ssrfGuard, timeoutGuard } from "../../security/hardening/index.js";
import { budgetedChunks, incrementalParse, lazyExtract, memoryBudget, parserPool, streamParse } from "../../performance/index.js";

export { budgetedChunks, decompressionGuard, enforceResourceBudget, incrementalParse, isSafeRemoteTarget, lazyExtract, malformedPayloadGuard, memoryBudget, memoryGuard, parserPool, recursionGuard, redirectGuard, sandboxText, ssrfGuard, streamParse, timeoutGuard };
